hid-uclogic.c

```diff
@@ -913,15 +980,19 @@ static int uclogic_enable_tablet(struct hid_device *hdev, __le16 **pbuf)
 	 * driver traffic.
 	 * NOTE: This enables fully-functional tablet mode.
 	 */
-	len = UCLOGIC_PRM_NUM * sizeof(*buf);
+	if (rdescid == UCLOGIC_HIRES_RDESC_ID) {
```

Comment:
This feels hacky and error-prone. I would rather pass the string descriptor buffer length along with the descriptor ID.